master/internal/api/socketlike.go

```diff
@@ -0,0 +1,28 @@
+package api
+
+import (
+	"fmt"
+
+	"github.com/gorilla/websocket"
+)
+
+type ReadWriter interface {
```

Comment:
this sounds too close to the standard library's `io.ReaderWriter`, but it isn't like the standard library's `ReaderWriter`.  I would prefer this be called `WebsocketLike` or anything that indicates it's intended to be a standing for a websocket.